[20026] Bump version to v2.6.7 (#623)
eProsima/Fast-DDS-docs: docs/notes/notes.rst

Information about the release lifecycle can be found
`here <https://github.com/eProsima/Fast-DDS/blob/master/RELEASE_SUPPORT.md>`_.

Version 2.6.7
=============

This release includes the following **features**:

> 1. Support `Autofill port` (:ref:`automatically set a port<transport_tcp_transportDescriptor>`) for TCP Transport.

Reviewer: In rst, quoting is done with two single-quote chars, applies to the whole list

```suggestion
1. Support ``Autofill port`` (:ref:`automatically set a port<transport_tcp_transportDescriptor>`) for TCP Transport.
```

2. Define a :ref:`super client<env_vars_ros_super_client>` by environment variable
3. Support :ref:`TCP Discovery server<use-case-tcp-discovery-server>` CLI and environment variable
4. Define methods (:ref:`environment variable<env_vars_builtin_transports>`,
   :ref:`rtps layer<rtps_layer_builtin_transports>`, :ref:`xml<RTPS>`) to
   :ref:`configure transport scenarios<transport_tcp_enabling>`
5. :ref:`Secure discovery server<DS_security>`

This release includes the following **improvements**:

> 1. Log warning message upon receiver resource creation failure.
> 2. Use foonathan memory manager for reducing allocations in `SharedMemManager.hpp`
> 3. Simplify code in `CDRMessage`.
> 4. Rerun failed tests with ctest option instead of colcon's.
> 5. Add CCache to all CI jobs.
> 6. Upgrade CMake minimum requirement to 3.16.3
> 7. Update PR checklist template. Backports and Description

Reviewer: ```suggestion
1. Log warning message upon receiver resource creation failure.
2. Add tests for reconnection with same GUID
3. Use foonathan memory manager for reducing allocations in ``SharedMemManager.hpp``
4. Simplify code in ``CDRMessage``.
5. Rerun failed tests with ctest option instead of colcon's.
6. Several improvements on CI jobs.
7. Upgrade CMake minimum requirement to 3.16.3
8. Update PR checklist template. Backports and Description
```

This release includes the following **bugfixes**:

1. Fix `DomainParticipant::register_remote_type` return when negotiating type.
2. Fix `RemoteBuiltinEndpointHonoring` blackbox test.
3. Allow participant profiles with no rtps tag.
4. Fix bad-free when receiving malformed DATA submessage.

> 5. Fix clang warnings (https://github.com/eProsima/Fast-DDS/pull/3906)

Reviewer: ```suggestion
5. Fix clang warnings
```

6. Use STL implementation of `Timed/RecursiveTimedMutex` when `MSVC >= 19.36`.
7. Fix encapsulation format in WLP.
8. Fix the clang build for clang 14.
9. Notify datasharing listener at the end of a successful matching in intraprocess.
10. Updatable disable_positive_acks period.
11. Fix Data Race when updating liveliness changed in WLP.
12. Fix TCP sender resources creation.

> 13. Add tests for reconnection with same GUID.

Reviewer: ```suggestion
```

14. Fix flow controllers utests compilation when using Fast CDR from thirdparty.
15. Add XML parser bit_bound bounds check.
16. Use `FASTRTPS_NO_LIB` on unittest root folder.
17. Use `SO_EXCLUSIVEADDRUSE` for Win32 unicast listening sockets.
18. Fix mac address overflow on windows.
19. Fix `PubSubAsReliable` test.
20. Fix `FileWatchTest`.
21. Add missing thread include.
22. Fix missing mandatory attribute check in XML parser struct type.
23. Better handling of trigger events in docs CI.

> 24. Fix memory problem when ciphering payload

Reviewer: Missing:
* Select correct .repos file on push events
* Update TLS unit test certificates
